lib/request-handler.js

```diff
@@ -63,94 +63,100 @@ module.exports = function processRequest (options, request, response) {
     };
 
     let shouldWriteHead = true;
-    let index = 0;
 
-    contextPromise.then((context) => {
+    const contentLengthStream = new ContentLengthStream((contentLength) => {
+        this.emit('end', request, contentLength);
+    });
 
-        const contentLengthStream = new ContentLengthStream((contentLength) => {
-            this.emit('end', request, contentLength);
-        });
+    const handleError = (err) => {
+        this.emit('error', request, err);
+        if (shouldWriteHead) {
+            shouldWriteHead = false;
+            let statusCode = 500;
+            if (err.code === TEMPLATE_NOT_FOUND) {
+                statusCode = 404;
+            }
 
-        const resultStream = new StringifierStream((tag) => {
-            const { placeholder, name, } = tag;
-            if (placeholder === 'pipe') {
-                return pipeDefinition(pipeInstanceName);
+            response.writeHead(statusCode, responseHeaders);
+            // To render with custom error template
+            if (typeof err.presentable === 'string') {
+                response.end(`${err.presentable}`);
+            } else {
+                response.end();
             }
+        } else {
+            contentLengthStream.end();
+        }
+    };
+
+    Promise.all([
+        templatePromise,
+        contextPromise
```

Comment:
an error in context should not control the page and return 500.

Reply:
There's a catch above.. Not super visible, though %)

Reply:
cool. missed it :)